js/models/Profile.js

```diff
@@ -0,0 +1,158 @@
+import _ from 'underscore';
+import BaseModel from './BaseModel';
+import app from '../app';
+import is from 'is_js';
+
+export default class extends BaseModel {
+  defaults() {
+    return {
+      primaryColor: '#086A9E',
+      secondaryColor: '#317DB8',
+      textColor: '#ffffff',
+      social: [],
+    };
+  }
+
+  url() {
+    // url is handled by sync, but backbone bombs if I don't have
+    // something explicitly set
+    return 'use-sync';
+  }
+
+  getColorFields() {
+    return [
+      'primaryColor',
+      'secondaryColor',
+      'textColor',
+    ];
+  }
+
+  // Ensure any colors are strings and have a leading hash.
+  standardizeColorFields(attrs = {}) {
+    const updatedAttrs = { ...attrs };
+
+    this.getColorFields().forEach((field) => {
+      if (typeof attrs[field] !== 'undefined') {
+        updatedAttrs[field] = updatedAttrs[field].toString();
+        updatedAttrs[field] = updatedAttrs[field].charAt(0) !== '#' ?
+          `#${updatedAttrs[field]}` : updatedAttrs[field];
+      }
+    });
+
+    return updatedAttrs;
+  }
+
+  set(key, val, options) {
+    // Handle both `"key", value` and `{key: value}` -style arguments.
+    let attrs;
+    let opts = options;
+
+    if (typeof key === 'object') {
+      attrs = key;
+      opts = val;
+    } else {
+      (attrs = {})[key] = val;
+    }
+
+    if (attrs.social) {
+      if (!_.isEqual(attrs.social, this._social)) {
+        this.attributes.social = this._social;
+
+        // if the object contents are not equal, we'll clone a new object so
+        // the change is event is fired.
+        this._social = JSON.parse(JSON.stringify(attrs.social)); // deep clone
+        attrs.social = JSON.parse(JSON.stringify(attrs.social));
+      }
+    }
+
+    return super.set(this.standardizeColorFields(attrs), opts);
+  }
+
+  get socialTypes() {
+    return [
+      'facebook',
+      'twitter',
+      'instagram',
+      'other',
+    ];
+  }
+
+  validate(attrs) {
```

Comment:
So basically if a field has an error we add an entry to an error object that we're returning from this method. The object is keyed by the field name and for each key there would be an array of errors (which might just be one). If a field has no errors, then it wouldn't be added to the object.

For example:

```
{
  'website': ['Please provide an url.'],
  'about': ['This must start with a lowercase q.', 'Some other error']
}
```

If we stay consistent with this approach, then this works hand in hand with our `formError.html` template which is available as template helper to all templates and is used to render the list of errors for a form field.

Anyhow, please see the example page of how to wire things together. It's just one approach, but I basically have the `model.validationError` contents passed into the template in `render`. That attribute is set by Backbone with the results of the most recent validation (e.g. if you call `model.save()` and client side validation fails, that attribute will contain the error object).